server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -288,6 +288,15 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Property.NodeScope
     );
 
+    // value 0 can disable dynamic pruning optimization in cardinality aggregation
+    public static final Setting<Integer> CARDINALITY_AGGREGATION_PRUNING_THRESHOLD = Setting.intSetting(
+        "search.dynamic_pruning.cardinality_aggregation_threshold",
```

Comment:
`search.optimize.cardinality_aggregation.max_field_cardinality_allowed`?

Reply:
`optimize` feels not that meaningful here. I intend to keep `dynamic_pruning`. We will have document for this setting too.
update to `search.dynamic_pruning.cardinality_aggregation.max_allowed_cardinality`

Reply:
I don't have a strong opinion here but given we have been doing these optimizations in search, so I thought maybe clubbing them together under `search.optimize` might make sense instead of these setting randomly present under different sections under `search` in `_cluster/settings`.

Reply:
I see. I treat this as a 2-way door. Considering the power user who is able to find and tune this setting, changing this settings later probably not a hard problem for them.